xsnap/sources/xsnap-worker.c

```diff
@@ -147,6 +147,67 @@ typedef enum {
 	E_TOO_MUCH_COMPUTATION = 17,
 } ExitCode;
 
+#define MAX_TIMESTAMPS 100
+static struct timeval timestamps[MAX_TIMESTAMPS];
```

Comment:
Global mutable state. Harrumph.

Reply:
I recently watched a [talk by Joe Armstrong](https://www.infoq.com/presentations/self-heal-scalable-system/) ... Erlang processes cost just a few hundred bytes each.... It reminded me of our situation to put more than one vat in a process.

But that's a looooong way from the top of our list, so I won't insist on passing this around explicitly today.

Reply:
Fair point. For the sake of practice, I'll sketch out what it would take to avoid this global state:

* define a `struct timestamps`
* declare an instance of it on the stack of `main()`, next to the `xsMachine *`
  * (in general I'm scared of declaring large structures on the stack, especially ones that get written to in variable amounts: if I had to choose between overwriting the stack and overwriting the heap, I'll pick the heap)
* define `resetTimestamps`, `recordTimestamp`, and `renderTimestamps` to accept a `struct timestamps *`
* change `fxWriteOkay` to accept a `struct timestamps *`
  * also accept a pointer to a buffer to fill, and change all callers of `fxWriteOkay` (all are in `main()`) to reference a rendering buffer declared on the stack of `main()`
  * (same concern about overwriting the stack)
  * or include the rendering buffer in `struct timestamps`
* change `xs_issueCommand` to accept `struct timestamps *` because it's needed by `recordTimestamp`
  * .. but `xs_issueCommand` is a callback invoked from the engine, so I can't really change its signature
  * so really, I'd need to modify `mxMachinePlatform` or something to jam the `struct timestamps *` into the `xsMachine` structure

Yep, I think that's not practical for today.